experimental/lib/Transforms/Speculation/HandshakeSpeculation.cpp

```diff
@@ -709,14 +718,14 @@ void HandshakeSpeculationPass::runDynamaticPass() {
   if (failed(placeUnits<handshake::SpecSaveOp>(this->specOp.getSaveCtrl())))
     return signalPassFailure();
 
-  // Place Commit operations
-  if (failed(placeCommits()))
-    return signalPassFailure();
-
   // Place SaveCommit operations and the SaveCommit control path
   if (failed(prepareAndPlaceSaveCommits()))
     return signalPassFailure();
 
+  // Place Commit operations
+  if (failed(placeCommits()))
+    return signalPassFailure();
+
```

Comment:
To fix the ordering of units when commit and save-commit are placed on the same edge